server/src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -960,6 +960,28 @@ public boolean doesConsumerExist(String uuid) {
         return result != 0;
     }
 
+    /**
+     * @param setOfUids of consumer UUIDs.
+     * @return set of consumer UUIDs that doesnt exists.
+     */
+    public Set<String> getNonExistingConsumer(Set<String> setOfUids) {
```

Comment:
This is still very much being defined and implemented as a specific-purpose function, but at the curator level (which is general purpose). If this is the way you want this method to work, define it in the resource as a private function and call the standard lookup functions. If this is to stay in the curator, you need to attempt to generalize it so that it can be used elsewhere without needing to be concerned with the requirements of the endpoint that leaked into its implementation.

That is:
- It still uses a slow lookup of full Consumer instances (can be replicated by ```getConsumers```)
- It's still requiring a specific collection type (Set)
- It's still modifying that input collection, which prevents use of several proxy collections and unmodifiable/singleton collections
- The output is superfluous since the input collection is modified directly
- The behavior of the input collection modification is still undocumented and only apparent by reading the code
- The method name implies something other than the behavior implemented


As touched on above, it's also fetching entire consumer instances when all the logic cares about is whether or not a given consumer UUID represents an existing consumer or not. This being the case, the query should be fetching only consumer UUIDs (or absent UUIDs).

Reply:
Input collection is not being modified now. Criteria query is fixed to fetch UUIDs only.